Classes/Service/DocumentService.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * (c) Kitodo. Key to digital objects e.V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo and TYPO3 projects.
+ *
+ * @license GNU General Public License version 3 or later.
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Kitodo\Dlf\Service;
+
+use Kitodo\Dlf\Domain\Repository\DocumentRepository;
+use TYPO3\CMS\Core\Log\LogManager;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use Kitodo\Dlf\Common\AbstractDocument;
+use Kitodo\Dlf\Domain\Model\Document;
+use TYPO3\CMS\Core\Utility\MathUtility;
+
+/**
+ * (Service For Abstract Controller) Service that decouples the loading of documents from each controller to only once.
+ *
+ * @package TYPO3
+ * @subpackage dlf
+ *
+ * @access public
+ */
+class DocumentService
+{
+    /**
+     * @access protected
+     * @var Document|null This holds the current document
+     */
+    protected ?Document $document = null;
+    /**
+     * @access protected
+     * @var array
+     */
+    protected $settings;
+    /**
+     * @access protected
+     * @var array
+     */
+    protected $logger;
+    protected DocumentRepository $documentRepository;
+
+    public function __construct()
+    {
+        $this->documentRepository = GeneralUtility::makeInstance(DocumentRepository::class);
+        $this->logger = GeneralUtility::makeInstance(LogManager::class)->getLogger(__CLASS__);
+    }
+
+    protected function init()
+    {
+        return true;
+    }
+    protected function reset()
+    {
+    }
+    /**
+     * @access public
+     * @param int $recordId
+     * @param array $settings
+     * @return ?Document
+     */
+    public function getDocument($recordId, $settings)
+    {
+        if ($this->document === null) {
+            $this->serviceLoadDocument($recordId, $settings);
+        }
+        return $this->document;
+    }
+    /**
+     * @access public
+     * @param int $recordId
+     * @param array $settings
+     */
+    private function serviceLoadDocument($recordId, $settings)
+    {
+        $this->settings = $settings;
+        // Get document ID from request data if not passed as parameter.
+        if (!empty($recordId)) {
+            $documentId = $recordId;
+        }
```

Comment:
This does not what the comment suggests. In fact, if the document ID is not passed as parameter, document loading will fail...
Please have a closer look at the original implementation!